### PR TITLE
Update CHANGELOG for 15.0.4 version bump

### PR DESCRIPTION
**Pull request by GoodForOneFare**
Still have #243 to merge, but this is a good start.

Please review @TylerHorth, @lemonmade

### CODE REVIEW

**Comment by lemonmade** [comment]
Wrong Tyler I think 😛

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,17 @@
 # Changelog
 
 ## [Unreleased]
+
+## [15.0.4]
```
Comment (lemonmade) [inline CHANGELOG.md]: I thought we did the ESLint update as 15.1.0? It seems like we have fallen out of sync here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 ### Added
-- Eslint rules: `no-return-await`, `no-useless-return`, and `func-name-matching`.
+- Updated all ESLint dependencies, which add the following rules: `no-return-await`, `no-useless-return`, and `func-name-matching`.
```
Comment (lemonmade) [inline CHANGELOG.md]: Just updated ESLint, none of the other deps were updated.

**Comment by GoodForOneFare** [comment]
I accidentally bumped `eslint-plugin-shopify` to 15.1.0 :/  Closing this in favour of a new 15.1.0 PR for esify/shopify-codemods.

**Pull request closed without merging**